pkg/Makefile: buglet passing TESTS

I think this delegation stuff is pretty hairy.

```bash
✔ ~/go/src/github.com/cockroachdb/cockroach/pkg [state-never-empty|✔]
07:49 $ make test PKG=./storage TESTS='a|b'
/bin/sh: b: command not found
make: *** [test] Error 127
```

--- a/build/teamcity-check.sh
+++ b/build/teamcity-check.sh
@@ -1,6 +1,7 @@
 #!/usr/bin/env bash
 set -exuo pipefail
 build/builder.sh make check 2>&1 | go-test-teamcity
+build/builder.sh make gotestdashi 2>&1
 build/builder.sh go generate ./... 2>&1
 build/builder.sh /bin/bash -c '! git status --porcelain | read || (git status; git diff -a; exit 1)' 2>&1
 

--- a/pkg/Makefile
+++ b/pkg/Makefile
@@ -12,6 +12,5 @@ DIRS=$(filter-out Makefile,$(wildcard *))
 
 .PHONY: $(DIRS)
 .DEFAULT $(DIRS):
-	@make -C .. $(MAKECMDGOALS) \
-	  $(patsubst PKG=%,PKG=./pkg/%,$(filter PKG=%,$(MAKEFLAGS))) \
-	  $(filter-out PKG=%,$(MAKEFLAGS))
+	@$(MAKE) -C .. $(MAKECMDGOALS) \
+	  $(patsubst PKG=%,PKG=./pkg/%,$(filter PKG=%,$(MAKEFLAGS)))

